apm-agent-plugins/apm-opentracing-plugin/src/main/java/co/elastic/apm/agent/opentracing/impl/ExternalSpanContextInstrumentation.java

```diff
@@ -0,0 +1,106 @@
+// * this work for additional information regarding copyright
+// * ownership. Elasticsearch B.V. licenses this file to you under
+// * the Apache License, Version 2.0 (the "License"); you may
+// * not use this file except in compliance with the License.
+// * You may obtain a copy of the License at
+// *
+// *   http://www.apache.org/licenses/LICENSE-2.0
+// *
+// * Unless required by applicable law or agreed to in writing,
+// * software distributed under the License is distributed on an
+// * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// * KIND, either express or implied.  See the License for the
+// * specific language governing permissions and limitations
+// * under the License.
+// * #L%
+// */
+//package co.elastic.apm.agent.opentracing.impl;
+//
+//import co.elastic.apm.agent.bci.VisibleForAdvice;
+//import co.elastic.apm.agent.impl.transaction.TraceContext;
+//import net.bytebuddy.asm.Advice;
+//import net.bytebuddy.description.method.MethodDescription;
+//import net.bytebuddy.description.type.TypeDescription;
+//import net.bytebuddy.implementation.bytecode.assign.Assigner;
+//import net.bytebuddy.matcher.ElementMatcher;
+//import org.slf4j.Logger;
+//import org.slf4j.LoggerFactory;
+//
+//import javax.annotation.Nullable;
+//import java.util.Collections;
+//import java.util.Map;
+//
+//import static net.bytebuddy.matcher.ElementMatchers.named;
+//
+//public class ExternalSpanContextInstrumentation extends OpenTracingBridgeInstrumentation {
+//
+//    private static final Logger logger = LoggerFactory.getLogger(ExternalSpanContextInstrumentation.class);
+//
+//    private final ElementMatcher<? super MethodDescription> methodMatcher;
+//
+//    public ExternalSpanContextInstrumentation(ElementMatcher<? super MethodDescription> methodMatcher) {
+//        this.methodMatcher = methodMatcher;
+//    }
+//
+//    @Override
+//    public ElementMatcher<? super TypeDescription> getTypeMatcher() {
+//        return named("co.elastic.apm.opentracing.ExternalProcessSpanContext");
+//    }
+//
+//    @Override
+//    public ElementMatcher<? super MethodDescription> getMethodMatcher() {
+//        return methodMatcher;
+//    }
+//
+//
+//    public static class ToTraceIdInstrumentation extends ExternalSpanContextInstrumentation {
+//
+//        public ToTraceIdInstrumentation() {
+//            super(named("toTraceId"));
+//        }
+//
+//
+//        @Advice.OnMethodExit(suppress = Throwable.class)
+//        public static void toTraceId(@Advice.FieldValue(value = "traceContext", typing = Assigner.Typing.DYNAMIC) @Nullable TraceContext traceContext,
+//                                     @Advice.Return(readOnly = false) String traceId) {
+//            if (traceContext != null) {
+//                traceId = doGetTraceId(traceContext);
+//            }
+//        }
+//
+//        @VisibleForAdvice
+//        public static String doGetTraceId(TraceContext traceContext) {
+//            return traceContext.getTraceId().toString();
```

Comment:
To get the id for `ExternalProcessSpanContext`, you could get the `TraceContext.TRACE_PARENT_HEADER` header from the `TextMap`, parse it via `TraceContext.asChildOf(String)` and return `context.getParentId()`.

Reply:
Where did I get the context? :(